### PR TITLE
Live reload for $ donejs develop. Closes #12

### LINKED ISSUE #12: Make live-reload work again
Currently, live-reload isn't working.

As this is a `donejs` app, we should make `donejs develop` start the node server, and live-reload correctly.

We aren't going to be using can-serve like [place-my-order does](https://github.com/donejs/place-my-order/blob/master/package.json#L15).

Instead, we need to probably need to add a script to [bitball's package.json](https://github.com/donejs/bitballs/blob/master/package.json#L7) that calls `node index.js` with an additional `--develop` option.  

`index.js` needs to be updated to start live-reload.  Here's how `can-serve` calls `steal-tools live-reload`: https://github.com/canjs/can-ssr/blob/master/bin/can-serve#L43.  My guess is that @James0x57 did something similar for RoomPlanner.  

I'm not a fan of having to spawn another process and having to depend on `infanticide` to kill it correctly.  Findout from @matthewp if there's a way to programmatically.   Perhaps a single node can only be one one process, hopefully not.

Bonus points if you can make this work in windows.  Everything will have to work in windows eventually.

### PR DESCRIPTION
**Pull request by James0x57**
Closes #12 

Note: I could not get bitballs to work on Windows because I couldn't get postgres to work from command line. Spent a few hours tonight trying to figure it out. Got past a couple problems but ran into a "role 'username' does not exist" problem that was a brick wall for now.

### CODE REVIEW

**Comment by akagomez** [comment]
>  Got past a couple problems but ran into a "role 'username' does not exist" problem that was a brick wall for now.

This may not be necessary to overcome. It wasn't in my case and thus wound up being a huge distraction. 

Is the client throwing that error?

**Comment by akagomez** [comment]
+1

**Comment by James0x57** [comment]
@akagomez The error was in git bash when I try to $ createdb bitballs. Never got to loading the client.

File: app.js [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+  //is dev mode so do live reload
+  var child = exec( "node_modules/.bin/steal-tools live-reload", {
+    cwd: process.cwd() + "/public"
+  });
+
+  child.stdout.pipe( process.stdout );
+  child.stderr.pipe( process.stderr );
+  
+  app.use( function ( req, res, next ) {
+    req.isDev = true;
```
Comment (justinbmeyer) [inline app.js]: please remove this.

**Pull request merged**